diff -Naur a/configure.py b/configure.py
--- a/configure.py	2019-11-19 08:29:05.000000000 +0000
+++ b/configure.py	2019-12-19 23:02:45.166199599 +0000
@@ -868,7 +868,7 @@
     '__MIPSEL__'  : 'mipsel',
     '__mips__'    : 'mips',
     '__PPC64__'   : 'ppc64',
-    '__PPC__'     : 'ppc64',
+    '__PPC__'     : 'ppc',
     '__x86_64__'  : 'x64',
     '__s390x__'   : 's390x',
   }
diff -Naur a/deps/v8/src/libsampler/sampler.cc b/deps/v8/src/libsampler/sampler.cc
--- a/deps/v8/src/libsampler/sampler.cc	2019-11-19 08:29:07.000000000 +0000
+++ b/deps/v8/src/libsampler/sampler.cc	2019-12-19 23:06:40.325604971 +0000
@@ -423,10 +423,17 @@
   state->lr = reinterpret_cast<void*>(ucontext->uc_mcontext.regs->link);
 #else
   // Some C libraries, notably Musl, define the regs member as a void pointer
+  #if !V8_TARGET_ARCH_32_BIT
   state->pc = reinterpret_cast<void*>(ucontext->uc_mcontext.gp_regs[32]);
   state->sp = reinterpret_cast<void*>(ucontext->uc_mcontext.gp_regs[1]);
   state->fp = reinterpret_cast<void*>(ucontext->uc_mcontext.gp_regs[31]);
   state->lr = reinterpret_cast<void*>(ucontext->uc_mcontext.gp_regs[36]);
+  #else
+  state->pc = reinterpret_cast<void*>(ucontext->uc_mcontext.gregs[32]);
+  state->sp = reinterpret_cast<void*>(ucontext->uc_mcontext.gregs[1]);
+  state->fp = reinterpret_cast<void*>(ucontext->uc_mcontext.gregs[31]);
+  state->lr = reinterpret_cast<void*>(ucontext->uc_mcontext.gregs[36]);
+  #endif
 #endif
 #elif V8_HOST_ARCH_S390
 #if V8_TARGET_ARCH_32_BIT
diff -Naur a/node.gyp b/node.gyp
--- a/node.gyp	2019-11-19 08:29:09.000000000 +0000
+++ b/node.gyp	2019-12-19 23:00:29.204078067 +0000
@@ -693,6 +693,11 @@
       'msvs_disabled_warnings!': [4244],
 
       'conditions': [
+        [ 'host_arch=="mips" or host_arch=="mipsel" or host_arch=="ppc"', {
+	  'link_settings': {
+	    'libraries': [ '-latomic' ],
+	  },
+	}],
         [ 'node_shared=="true"', {
           'sources': [
             'src/node_snapshot_stub.cc',
